File: frontend/src/app/models/chat-room.ts
export interface RegisteredUser {
  id: number;
  username: string;
  email: string;
}

export interface ChatMessage {
  id: number;
  message: string;
  sender: string;
  creationDate: string;
  messageType: string;
  roomId: string;
}

export interface ChatRoom {
  id: number;
  name: string;
  description: string;
  ownerId: string;
  registeredUsers: RegisteredUser[];
  chatMessages: ChatMessage[];
}

File: chat-app/src/main/java/dev/al/internship/chatapp/model/entity/ChatRoom.java
package dev.al.internship.chatapp.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class ChatRoom {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private String ownerId;

    @ManyToMany
    @JoinTable(
            name = "chat_room_users",
            joinColumns = @JoinColumn(name = "room_id"),
            inverseJoinColumns = @JoinColumn(name = "user_id")
    )
    private Set<User> registeredUsers;

    @OneToMany(mappedBy = "roomId", cascade = CascadeType.ALL)
    private Set<ChatMessage> chatMessages;
}
